### PR TITLE
API Revisions

### PR DESCRIPTION
**Pull request by ronenh**
This PR includes breaking changes! It would be a minor version bump.o

* Remove `SessionID`
* Remove `DialOptionsProvider` and roll its functionality into `ConnectionOptions`.
* Embed `Config` in `ConnectionOptions` to eliminate duplication.
* Consolidate the logic that creates gRPC dial options into `ConnectionOptions`.
* Support mTLS using `WithClientCert()`.
* Support `WithAccountID` and its equivalent config option.
* Support `WithNoTLS` and its equivalent config option.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -1,28 +1,85 @@
...
+	// Token and APIKey are mutually exclusive.
+	Token string `json:"token"`
+
+	// An API key to be used for authentication with the service.
+	APIKey string `json:"api_key"`
+
+	// An Aserto tenant ID.
+	TenantID string `json:"tenant_id"`
+
+	// An aserto account ID.
```
Comment (gertd) [inline config.go:30]: Aserto (uppercase)

File: ds/v2/directory.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ func New(opts ...aserto.ConnectionOption) (*Client, error) {
 		return nil, err
 	}
 
-	if options.ServerAddress() == "" {
+	if options.Address == "" {
```
Comment (gertd) [inline ds/v2/directory.go:39]: Do we still need to v2 directory?

File: config.go [hunk 3]
```diff
@@ -1,28 +1,85 @@
...
+}
+
+// Converts the Config into a ConnectionOption slice that can be passed to NewConnection().
+func (cfg *Config) ToConnectionOptions() ([]ConnectionOption, error) {
 	if cfg.APIKey != "" && cfg.Token != "" {
-		return nil, errors.New("both api_key and token are set")
+		return nil, errors.Wrap(ErrInvalidConfig, "api_key and token are mutually exclusive")
+	}
+
+	if cfg.Insecure && cfg.NoTLS {
```
Comment (gertd) [inline config.go:72]: Do we need a mutually exclusive check for mTLS and (insecure | NoTLS)?

File: ds/v2/directory.go (hunk 2 above)
Comment (ronenh) [reply ds/v2/directory.go:39]: You're right. I'll get rid of it.

File: config.go (hunk 3 above)
Comment (ronenh) [reply config.go:72]: Good catch!

**Review by gertd — CHANGES_REQUESTED** [review]
See comments inline
* Need for directory v2, I would propose removing it, as topaz 32 no longer supports, and this go-aserto release will be a major version rev.
* Need for mTLS mutually exclusive check with Insecure or  NoTLS

**Review by gertd — APPROVED** [review]

**Pull request merged**